apprise/plugins/NotifyStreamlabs.py

```diff
@@ -0,0 +1,278 @@
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions :
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+
+# For this to work correctly you need to register an app
+# and generate an access token
+#
+#
+#  This plugin will simply work using the url of:
+#     streamlabs://access_token/
+#
+# API Documentation on Webhooks:
+#    - https://dev.streamlabs.com/
+#
+import requests
+
+from .NotifyBase import NotifyBase
+from ..common import NotifyType
+from ..utils import validate_regex
+from ..AppriseLocale import gettext_lazy as _
+
+
+class NotifyStreamlabs(NotifyBase):
+    """
+    A wrapper to Streamlabs Donation Notifications
+
+    """
+    # The default descriptive name associated with the Notification
+    service_name = 'Streamlabs'
+
+    # The services URL
+    service_url = 'https://streamlabs.com/'
+
+    # The default secure protocol
+    secure_protocol = 'strmlabs'
+
+    # A URL that takes you to the setup/help of the specific protocol
+    setup_url = 'https://github.com/caronc/apprise/wiki/Notify_streamlabs'
+
+    # Streamlabs Api endpoint
+    notify_url = 'https://streamlabs.com/api/v1.0/donations'
+
+    # Allows the user to specify the NotifyImageSize object
+    image_size = None
+
+    # The maximum allowable characters allowed in the body per message
+    body_maxlen = 255
+
+    # Define object templates
+    templates = (
+        '{schema}://{access_token}/',
+    )
+
+    # Define our template tokens
+    template_tokens = dict(NotifyBase.template_tokens, **{
+        'access_token': {
+            'name': _('Access Token'),
+            'private': True,
+            'required': True,
+            'type': 'string',
+            'regex': (r'^[A-Za-z0-9]{40}$', 'i')
+        },
+    })
+
+    # Define our template arguments
+    template_args = dict(NotifyBase.template_args, **{
+        'amount': {
+            'name': _('Amount'),
+            'type': 'int',
+            'default': 0,
+            'min': 0
+        },
+        'currency': {
+            'name': _('Currency'),
+            'type': 'string',
+            'default': 'USD',
+            'regex': (r'^[A-Z]{3}$', 'i'),
+        },
+        'name': {
+            'name': _('Name'),
+            'type': 'string',
+            'default': 'Anon',
+            'regex': (r'^[A-Za-z0-9]{2,25}$', 'i')
```

Comment:
This regular expression somewhat implies that there can't be a space (` `) in the user's name.   Looking [here](https://dev.streamlabs.com/reference#donations), that doesn't seem to be an issue.  It just states:
>The name of the donor. has to be between 2-25 length and should only contain utf8 characters

Maybe just change the regular expression to read `r'[^\s].{1,24}` (has to start with something that isn't a space) and still enforces your `2-25` limit restriction.?